mono/metadata/mempool.c

```diff
@@ -95,6 +96,14 @@ mono_mempool_new (void)
  * \param initial_size the amount of memory to initially reserve for the memory pool.
  * \returns a new memory pool with a specific initial memory reservation.
  */
+// clang's ThreadSanitizer detects races of total_bytes_allocated and pool->d.allocated throughout the functions
```

Comment:
You can put this comment as part of the above comment in the following manner:

```
/**
 * mono_mempool_new_size:
 *
 *   clang's ThreadSanitizer detects races of total_bytes_allocated and pool->d.allocated throughout the functions
 *     * mono_mempool_alloc
 *     * mono_mempool_new_size
 *     * mono_mempool_destroy
 *   while these races could lead to wrong values, total_bytes_allocated is just used for debugging / reporting and since
 *   the mempool.c functions are called quite often, a discussion led the the conclusion of ignoring these races:
 *   https://bugzilla.xamarin.com/show_bug.cgi?id=57936
 *
 * \param initial_size the amount of memory to initially reserve for the memory pool.
 * \returns a new memory pool with a specific initial memory reservation.
 */
```